import os, sys

here = os.path.dirname(__file__)
sys.path.append(os.path.join(here, '..'))

import lrs_pb2, lrs_pb2_grpc
import logging
import grpc

def run():
    with grpc.insecure_channel("localhost:50052") as channel:
        stub = lrs_pb2_grpc.RoadNetworkStub(channel)
        routes_req = lrs_pb2.DownloadRequests(routes=[
            '01047', '01020', '0102614', '01048', '01055', '0102411', '0101921', '0104815', 
            '0100211', '010091', '010101', '01025', '0101922', '010051', '01029', '010361', 
            '0104811', '0101221', '0101225', '0102511', '0102913', '01035', '01045', '01013', 
            '0101311', '010302', '01042', '010431', '01001', '0100212', '010052', '0101121', 
            '01031', '0104812', '010123', '01015', '0102612', '010341', '0101211', '01044', 
            '0102613', '01051', '010112', '01024', '0103421', '01050', '010371', '0100111', 
            '01006', '010081', '01022', '010541', '0100311', '0101222', '0102912', '010301', 
            '0103011', '0103012', '0103511', '0100213', '01004', '01027', '0100113', '010122', 
            '01003', '01053', '01033', '010432', '0105011', '010191', '010192', '0102615', 
            '01002', '010071', '010121', '01021', '0102916', '0103512', '010542', '010102', 
            '0100214', '0100721', '010094', '0104011', '0104814', '010381', '0100112', '010082', 
            '010111', '01023', '01039', '0104813', '010072', '0101223', '01026', '01028', '010342', 
            '01052', '01041', '0104911', '0100411', '0103711', '0102611', '01032', '0104312', 
            '010543', '010053', '010092', '0101224', '010362', '010372', '01046', '010093', 
            '01018', '010382', '0105012'
        ], output_shp='SegmenRuas_Prov_01.shp')

        file_path = stub.DownloadAsSHP(routes_req)
        print(f"Request: {routes_req}, Response: {file_path}")

if __name__ == '__main__':
    logging.basicConfig()
    run()